Update suite contributors to respect build profiles

orangejenny opened the pull request:
https://dimagi-dev.atlassian.net/browse/ICDS-507

Looks like a substantial part of suite generation ignores the build profile id, which is probably fine most of the time, but it can be a problem when a menu's image/audio is present for some languages in the app but not for any of the languages in the build profile. Then the suite references the multimedia, but the app strings files, which do pay attention to the build profile, don't have a translation for it and the app errors at run time.

Opening for review and tests; I'm planning to add more tests and also think some more about how understandable the "for_default" logic is. Open to suggestions.

Feature flag (or privilege? I forget how access to these is controlled): build profiles

orangejenny committed:
Added logic to uses_(image|audio) to respect build profile id

orangejenny committed:
Added passing tests for suite, failing tests for app strings

orangejenny committed:
Added build_profile_id to _get_media_by_language

orangejenny committed:
Fixed tests and lint errors

emord commented:
How hard would it be to abstract build_profiles into actual builds? In other words, we already have the convention where a `build` is an `app`. Could we have something before the the builds with build profiles, are actually multiple builds where each profile is its own build?

orangejenny commented:
@emord let's offline

orangejenny committed:
Added more tests for app strings.

These aren't directly related to this PR, they're just to improve test coverage.

orangejenny commented:
Ready for review.

orangejenny committed:
Fixed lint error
